internal/xds/cache/snapshotcache.go

```diff
@@ -313,10 +306,14 @@ func (s *snapshotCache) OnDeltaStreamClosed(streamID int64, node *corev3.Node) {
 	delete(s.streamIDNodeInfo, streamID)
 	delete(s.deltaStreamDuration, streamID)
 
+	s.clearNodeSnapShot(node)
+}
+
+func (s *snapshotCache) clearNodeSnapShot(node *corev3.Node) {
 	s.nodeFrequency[node.Id] -= 1
 	if s.nodeFrequency[node.Id] <= 0 {
 		delete(s.nodeFrequency, node.Id)
-
+		delete(s.lastSnapshot, node.Cluster)
```

Comment:
Should we delete the snapshot for a irKey ?  Other nodes from the same irKey may still need it.